File: LeetCode/java/991.java
class Solution {
    public int brokenCalc(int X, int Y) {
        if (X >= Y)
            return X - Y;
        return Y % 2 == 1 ? brokenCalc(X, Y + 1) + 1 : brokenCalc(X, Y / 2) + 1;
    }
}

File: LeetCode/golang/991.go
func brokenCalc(X int, Y int) int {
    if X >= Y {
        return X - Y
    }
    if Y % 2 == 1 {
        return brokenCalc(X, Y + 1) + 1
    }
    return brokenCalc(X, Y / 2) + 1
}